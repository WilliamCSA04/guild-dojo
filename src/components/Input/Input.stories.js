import React from "react";
import Input from ".";
import { ThemeProvider } from "styled-components";
import theme from "../../styles";

export default {
  title: "Input"
};

export const Default = () => (
  <ThemeProvider theme={theme}>
    <Input ariaLabel="Default story" />
  </ThemeProvider>
);
