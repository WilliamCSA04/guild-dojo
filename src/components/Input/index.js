import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
const DefaultInput = styled.input.attrs(props => ({
  "data-testid": props.dataTestID
}))`
  border-radius: 0.2rem;
  border: 1px transparent solid;
  &:focus {
    border: 1px ${props => props.theme.colors.secondary} solid;
    box-shadow: none;
  }
  padding: 0.1rem 0.5rem;
  -moz-appearance: textfield;
  &::-webkit-inner-spin-button {
    -webkit-appearance: none;
  }
`;
export default function Input({
  id,
  required,
  className,
  onChange,
  autoFocus,
  type,
  ariaLabel,
  ariaLabelledby,
  defaultValue,
  placeholder,
  dataTestID
}) {
  return (
    <DefaultInput
      id={id}
      required={required}
      onChange={e => onChange(e.currentTarget.value)}
      className={className}
      autoFocus={autoFocus}
      type={type}
      dataTestID={dataTestID}
      ariaLabel={ariaLabel}
      ariaLabelledby={ariaLabelledby}
      defaultValue={defaultValue}
      placeholder={placeholder}
    />
  );
}

Input.propTypes = {
  ariaLabel: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  dataTestID: PropTypes.string.isRequired,
  autoFocus: PropTypes.bool,
  required: PropTypes.bool,
  id: PropTypes.string,
  ariaLabelledby: PropTypes.string,
  type: PropTypes.string,
  defaultValue: PropTypes.string,
  placeholder: PropTypes.string
};

Input.defaultProps = {
  required: false,
  ariaLabelledby: "",
  autoFocus: false,
  type: "text",
  defaultValue: "",
  placeholder: ""
};
