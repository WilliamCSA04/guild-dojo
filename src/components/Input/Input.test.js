import React from "react";
import { render } from "@testing-library/react";
import Input from ".";

describe("<Input /> snapshot", () => {
  it("renders the component with default props", () => {
    const container = render(
      <Input ariaLabel="Testing input" dataTestID="Test" onChange={() => {}} />
    );
    expect(container.asFragment()).toMatchSnapshot();
  });

  it("renders the component with defaultValue", () => {
    const container = render(
      <Input
        ariaLabel="Testing input"
        dataTestID="Test"
        onChange={() => {}}
        defaultValue="Text"
      />
    );
    expect(container.asFragment()).toMatchSnapshot();
  });
});
