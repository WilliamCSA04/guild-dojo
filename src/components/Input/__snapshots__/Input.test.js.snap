// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> snapshot renders the component with default props 1`] = `
<DocumentFragment>
  <input
    class="sc-AxjAm jmOayL"
    data-testid=""
    placeholder=""
    type="text"
    value=""
  />
</DocumentFragment>
`;

exports[`<Input /> snapshot renders the component with defaultValue 1`] = `
<DocumentFragment>
  <input
    class="sc-AxjAm jmOayL"
    data-testid=""
    placeholder=""
    type="text"
    value="Text"
  />
</DocumentFragment>
`;
